engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/cache/CachingDataFactory.java

```diff
@@ -360,7 +360,7 @@ private DataCacheKey createCacheKey( final String query, final DataRow parameter
       DataFactoryMetaData metaData = backend.getMetaData();
 
       final String[] referencedFields = metaData.getReferencedFields( backend, query, parameters );
-      if ( referencedFields != null ) {
+      if ( referencedFields != null && referencedFields.length > 0 ) {
```

Comment:
This is not correct. With this change you now would no longer cache any unparametrized query at all. Not caching that expensive "SELECT * WHERE x LIKE "%Y%" is going to be bad. Revert this line here.

(And yes, this now means that static table data sources are ending up in the cache, where they may not need to be cached (as they are in memory anyway), but that is better than not caching queries at all. We could address that as a extra setting in the metadata at some future point, but for today, screw it, better be correct than memory efficient.)

Reply:
Ok, just added because some assert on unit tests are failing due to this fact. Should I change test asserts instead?